From: Debian Multimedia Maintainers <debian-multimedia@lists.debian.org>
Date: Wed, 21 Aug 2024 10:22:40 +0200
Subject: Fix callback signatures and pointer-types

Origin: upstream
Applied-Upstream: 2a071ae8ca89bc21b5c80037f8c95a01bb670ac9, 596667daba1ed99eda048e491ff8f36200f09429
Last-Update: 2024-08-21

gcc>=14 is very picky about correct pointer types
Last-Update: 2024-08-21
---
 Opcodes/OSC.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Opcodes/OSC.c b/Opcodes/OSC.c
index 529e7b7..27dbae8 100644
--- a/Opcodes/OSC.c
+++ b/Opcodes/OSC.c
@@ -483,7 +483,7 @@ static int32_t OSCcounter(CSOUND *csound, OSCcount *p)
 }
 
 static int32_t OSC_handler(const char *path, const char *types,
-                       lo_arg **argv, int32_t argc, void *data, void *p)
+                       lo_arg **argv, int32_t argc, lo_message data, void *p)
 {
     IGN(argc);  IGN(data);
     OSC_PORT  *pp = (OSC_PORT*) p;
@@ -548,7 +548,7 @@ static int32_t OSC_handler(const char *path, const char *types,
             case 'b':
               {
                 int32_t len =
-                  lo_blobsize((lo_blob*)argv[i]);
+                  lo_blobsize((lo_blob)argv[i]);
                 m->args[i].blob =
                   csound->Malloc(csound,len);
                 memcpy(m->args[i].blob, argv[i], len);
@@ -964,7 +964,7 @@ static int32_t OSC_list(CSOUND *csound, OSCLISTEN *p)
 /* ******** ARRAY VERSION **** EXPERIMENTAL *** */
 
 static int32_t OSC_ahandler(const char *path, const char *types,
-                       lo_arg **argv, int32_t argc, void *data, void *p)
+                       lo_arg **argv, int32_t argc, lo_message data, void *p)
 {
     IGN(argc);  IGN(data);
     OSC_PORT  *pp = (OSC_PORT*) p;
